### ISSUE TITLE
onnx and pytorch get different inference results

### ISSUE THREAD

**Issue by Whoo-jl** [SysCV/sam-hq#50]
What factors are responsible for the phenomenon that onnx and pytorch get different inference results?

**Comment by ymq2017** [comment]
Hi, could you give a more detailed description of your input and some hyper-parameters?

**Comment by Whoo-jl** [comment]
> Hi, could you give a more detailed description of your input and some hyper-parameters?

test image: 
![example0](https://github.com/SysCV/sam-hq/assets/52949064/e7822ec3-10a2-46a4-8e56-9a6a0f2e80d2)

box prompt: [4,13,1007,1023]

model type: vit_l

pytorch inference demo:  [colab notebook](https://colab.research.google.com/drive/1QwAbn5hsdqKOD5niuBzuqQX4eLCbNKFL?usp=sharing) 
pytorch inference result:
![mask](https://github.com/SysCV/sam-hq/assets/52949064/df9459ea-91ce-4d57-961e-ffd3d3871c64)


onnx inferense demo: [example notebook](https://colab.research.google.com/drive/11U2La49c2IxahzJkAV-EzPqEH3cz_5hq?usp=sharing) 
onnx inference result:
![mask_onnx](https://github.com/SysCV/sam-hq/assets/52949064/d566957c-64ea-4add-baae-65803ee52360)

**Comment by ymq2017** [comment]
Hi, in the ONNX notebook, a parameter **hq_token_only** needs to be defined when exporting the ONNX model. In this notebook, we use the example
`onnx_model = SamOnnxModel(sam, hq_token_only=True)`
If you set **hq_token_only=False**, it will have the same result as torch.

**Comment by Whoo-jl** [comment]
> Hi, in the ONNX notebook, a parameter **hq_token_only** needs to be defined when exporting the ONNX model. In this notebook, we use the example `onnx_model = SamOnnxModel(sam, hq_token_only=True)` If you set **hq_token_only=False**, it will have the same result as torch.

It was indeed this parameter that was causing the problem, and I've located the problem, thanks for your reply!
